appshell/appshell_extensions_platform.h

```diff
@@ -84,6 +85,36 @@ inline int32 InstallCommandLineTools() { return ERR_CL_TOOLS_NOTSUPPORTED; }
 #endif
 
 
+
+
+inline void GetCharsetMatch(const char* bufferData, size_t bufferLength, std::string &detectedCharSet) {
```

Comment:
It is not advisable to add huge code inside header file. Can you add another file like `appshell_extension_platform.cpp` and add this definition there?